File: objc/Adapter/Adapter/EthernetConnection.m
#import "EthernetConnection.h"
#include <stdlib.h>

@implementation EthernetConnection

-(void)makeReady {
    NSLog(@"Making Ethernet Ready...");
}

-(BOOL)isReady {
    return (arc4random_uniform(100) % 2);
}

-(void)sendPacket:(NSString*)packet {
    if(packet.length > 10) return;
    NSLog(@"Packet sent. [%@]", packet);
}

@end

File: swift/Adapter/Adapter/EthernetConnection.swift
import Foundation

class EthernetConnection : Connection {
    
    init() {
        
    }
    
    func makeReady() {
        print("Making Ethernet Ready...")
    }
    
    func isReady() -> Bool {
        return arc4random_uniform(2) == 0
    }
    
    func sendPacket(_ data: String) {
        if data.count > 10 { return }
        print("Packet sent. [\(data)]")
    }
}
